Workspace/apps/admin-dashboard/test/functional/adminDashboard.js

```diff
@@ -0,0 +1,216 @@
+            // ComponentQuery - 'toolbar #main-navigation-btn'
+            // xpath - '//a[@id="main-navigation-btn"]'
+            return ST.component('#main-navigation-btn');
+        },
+        menuItem: function(itemName) {
+            // Another example of ComponentQuery used to lookup Ext.list.TreeItem component by xtype and text property
+            return ST.component('treelistitem[text='+itemName+']');
+        },
+        toolbarItem: function(itemName) {
+            // ComponentQuery using href property to locate right button in Top toolbar.
+            return ST.button('toolbar button[href=#'+itemName+']');
+        },
+        // Following locators are in test used to verify if correct view is loaded in application.
+        emailView: function() {
+            return ST.component('email');
+        },
+        profileView: function(){
+            return ST.component('profile');
+        },
+        searchView: function() {
+            return ST.component('searchresults');
+        },
+        blankPageView: function() {
+            return ST.component('pageblank');
+        },
+        loginView: function() {
+            return ST.component('login');
+        },
+        isDesktop : ST.os.deviceType == "Desktop"
+
+    };
+    beforeEach(function(){
+        Admin.app.redirectTo("#dashboard"); // make sure you are on dashboard homepage
+    });
+
+    describe("Example loads correctly", function(){
+        it("Admin dashboard page screenshot should match baseline", function(done) {
+            // Screenshots are only supported when running tests via CLI test runner
+            ST.screenshot('dash-navigation',done);
+        }, 1000 * 20);
+    });
+
+    describe('App navigation', function(){
+        describe("Tree menu", function(){
+            it("treelist menu should be initially expanded", function(){
+                Dash.treeList()
+                    // Waiting is needed in this case to make sure test won't progress before all necessary parts are in place
+                    .visible()
+                    .and(function(navMenu){
+                        // If Tree List is expanded micro should be null.
+                        expect(navMenu.getMicro()).toBeFalsy();
+                    });
+            });
+            it("clicking on hamburger should toggle treelist navigation mode - collapse", function() {
+                Dash.hamburger()
+                    .visible()
+                    .click();
+                Dash.treeList()
+                    .and(function(navMenu){
+                        expect(navMenu.getMicro()).toBe(true);
+                    });
+                //return to initial state
+                Dash.hamburger().click();
+            });
+            it("clicking on hamburger should toggle treelist navigation mode - collapse and expand", function(){
+                Dash.hamburger()
+                    .visible()
```

Comment:
The `visibility` state method should seldom be needed since input injection automatically waits for this.